types/pb/sf/ethereum/type/v2/type.go

```diff
@@ -256,6 +275,155 @@ func (block *Block) NormalizeInPlace() {
 
 }
 
+// CombinePolygonSystemTransactions will identify transactions that are "system transactions" and merge them into a single transaction with a predictive name, like the `bor` client does.
+// It reorders the calls and logs to match expected output from RPC API.
+func CombinePolygonSystemTransactions(traces []*TransactionTrace, blockNum uint64, blockHash []byte) (out []*TransactionTrace) {
+
+	var systemTransactions []*TransactionTrace
+	var normalTransactions []*TransactionTrace
```

Comment:
Let's pre-allocate this one to the number of total transaction:
```suggestion
	normalTransactions := make([]*TransactionTrace, 0, len(traces))
```